test/components/interactive/xDragBoxLayerTests.ts

```diff
@@ -0,0 +1,73 @@
+///<reference path="../../testReference.ts" />
+
+var assert = chai.assert;
+
+describe("Interactive Components", () => {
+  describe("XDragBoxLayer", () => {
+    var SVG_WIDTH = 400;
+    var SVG_HEIGHT = 400;
+
+    it("bounds()", () => {
+      var svg = generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      var dbl = new Plottable.Component.Interactive.XDragBoxLayer();
+      dbl.boxVisible(true);
+      dbl.renderTo(svg);
+
+      var topLeft = {
+        x: SVG_WIDTH / 4,
+        y: SVG_HEIGHT / 4
+      };
+      var bottomRight = {
+        x: SVG_WIDTH / 2,
+        y: SVG_HEIGHT / 2
+      };
+
+      dbl.bounds({
+        topLeft: topLeft,
+        bottomRight: bottomRight
+      });
+
+      var actualBounds = dbl.bounds();
+      assert.strictEqual(actualBounds.topLeft.y, 0, "box starts at top");
+      assert.strictEqual(actualBounds.topLeft.x, topLeft.x, "left edge set correctly");
+      assert.strictEqual(actualBounds.bottomRight.y, dbl.height(), "box ends at bottom");
+      assert.strictEqual(actualBounds.bottomRight.x, bottomRight.x, "right edge set correctly");
+
+      svg.remove();
+    });
+
+    it("resizes only in x", () => {
+      var svg = generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      var dbl = new Plottable.Component.Interactive.XDragBoxLayer();
+      dbl.boxVisible(true);
+      dbl.resizable(true);
+      dbl.renderTo(svg);
+
+      var topLeft = {
+        x: SVG_WIDTH / 4,
+        y: SVG_HEIGHT / 4
+      };
+      var bottomRight = {
+        x: SVG_WIDTH / 2,
+        y: SVG_HEIGHT / 2
+      };
+
+      dbl.bounds({
+        topLeft: topLeft,
+        bottomRight: bottomRight
+      });
+
+      var actualBounds = dbl.bounds();
+      var dragTo = {
+        x: SVG_WIDTH * 3 / 4,
+        y: SVG_HEIGHT / 2
+      };
+      var target = dbl.background();
+      triggerFakeDragSequence(target, actualBounds.bottomRight, dragTo);
+      actualBounds = dbl.bounds();
+      assert.strictEqual(actualBounds.bottomRight.x, dragTo.x, "resized in x");
+      assert.strictEqual(actualBounds.bottomRight.y, dbl.height(), "height did not change");
```

Comment:
the test message does not actually suggest what is being tested.

The test currently reads, "assert that the bottom of the bounds is at the bottom of the layer after this dragTo".  We will probably want this test but this is the wrong message for it.

We do however want the test with the message "height did not change", in which case, you should find the oldHeight before the dragSequence (using bounds.bottom.y - bounds.top.y) and compare it to the height after the dragSequence and then assert these 2 are the same.